src/jit/gentree.h

```diff
@@ -4258,6 +4258,8 @@ struct GenTreeDynBlk : public GenTreeBlk
     GenTreeDynBlk(GenTreePtr addr, GenTreePtr dynamicSize)
         : GenTreeBlk(GT_DYN_BLK, TYP_STRUCT, addr, 0), gtDynamicSize(dynamicSize), gtEvalSizeFirst(false)
     {
+        // Conservatively the 'addr' could be null or point into the global heap.
+        gtFlags |= GTF_EXCEPT | GTF_GLOB_REF;
```

Comment:
It may be worth checking to see if `addr` is a lclVar address, at which point we could omit these.